src/main/java/com/dropbox/core/v2/callbacks/DbxGlobalCallbackFactory.java

```diff
@@ -0,0 +1,17 @@
+package com.dropbox.core.v2.callbacks;
+
+/**
+ * Client should implement methods in this class to allow for consistent, global handling of route-specific error types,
+ * as well as general network errors. Normally, error handling is done on a request-by-request basis.
+ * However, it is convenient to handle some error behavior consistently, regardless of the request or the endpoint.
+ * For implementing global error handling for general networking errors like an auth error, `createNetworkErrorCallback`
+ * should be implemented. For implementing global error handling for route-specific errors, the
+ * `createRouteErrorCallback` should be implemented.
+ */
+public interface DbxGlobalCallbackFactory {
+    // Should instantiate separate callback object on each method invocation
+    <T> DbxRouteErrorCallback<T> createRouteErrorCallback(T routeError);
```

Comment:
This is good intuition, and what I originally tried to implement. However, Java strangely doesn't do dynamic overloaded method resolution at the parameter level, only the object level.

Read more [here](http://stackoverflow.com/questions/1572322/overloaded-method-selection-based-on-the-parameters-real-type).

tl;dr the parameter type only influences the overloaded method resolution at compile time, not run time. Given that the route error is generic at compile time, it always resolves to the declared generic method (not the more specific `ListFolderError` method).

Therefore, I've opted to keep the existing implementation, because I can't think of a better way of doing it.